### ISSUE TITLE
Get intermediate values from forward pass

### ISSUE THREAD

**Issue by fabiannagel** [google/jax#4857]
Is there a simple way to retrieve values from the forward pass when calling a compiled `grad()` function?

Say, I directly obtain the gradient like this:
```
def run_iteration(params, x, y):
  pred = forward_pass(params, x)
  return compute_loss(pred, y)

loss_grad = grad(run_iteration)
```

How would I retrieve e.g. the scalar loss? It feels like an obvious thing to do, but I couldn't find anything in the documentary.

**Comment by tomhennigan** [comment]
For this specific case you want `value_and_grad(f)`. If you want to return arbitrary intermediate values you want `value_and_grad(f, has_aux)` (or `grad`).

https://jax.readthedocs.io/en/latest/jax.html#jax.value_and_grad
